src/main/java/org/apache/commons/dbutils/BeanProcessor.java

```diff
@@ -65,19 +65,21 @@
      */
     private static final Map<Class<?>, Object> primitiveDefaults = new HashMap<Class<?>, Object>();
 
-    /**
-     * ServiceLoader to find <code>ColumnHandler</code> implementations on the classpath.  The iterator for this is
-     * lazy and each time <code>iterator()</code> is called.
-     */
-    // FIXME: I think this instantiates new handlers on each iterator() call. This might be worth caching upfront.
-    private static final ServiceLoader<ColumnHandler> columnHandlers = ServiceLoader.load(ColumnHandler.class);
+    private static final List<ColumnHandler> columnHandlers = new ArrayList<ColumnHandler>();
 
-    /**
-     * ServiceLoader to find <code>PropertyHandler</code> implementations on the classpath.  The iterator for this is
-     * lazy and each time <code>iterator()</code> is called.
-     */
-    // FIXME: I think this instantiates new handlers on each iterator() call. This might be worth caching upfront.
-    private static final ServiceLoader<PropertyHandler> propertyHandlers = ServiceLoader.load(PropertyHandler.class);
+    static{
+        for (ColumnHandler h : ServiceLoader.load(ColumnHandler.class)) {
+            columnHandlers.add(h);
+        }
+    }
+
+    private static final List<PropertyHandler> propertyHandlers = new ArrayList<PropertyHandler>();
+
+    static {
+        for (PropertyHandler h : ServiceLoader.load(PropertyHandler.class)) {
+            propertyHandlers.add(h);
```

Comment:
What not use a single static block?

Reply:
Can you explain why that is problematic?

Reply:
I think of it the other way around: Why use two blocks when you could use one?

Reply:
I didn't see there was already a static block initializing `primitiveDefaults`. I find it more readable initializing each static variable in their own block. I did an update and put all in one static block.